frontend/src/features/properties/list/PropertyListView.tsx

```diff
@@ -490,113 +501,135 @@ const PropertyListView: React.FC = () => {
-            <Table<IProperty>
-              name="propertiesTable"
-              lockPageSize={true}
-              columns={columns}
-              data={data || []}
-              loading={data === undefined}
-              filterable
-              sort={sorting}
-              pageIndex={pageIndex}
-              onRequestData={handleRequestData}
-              pageCount={pageCount}
-              onSortChange={(column: string, direction: SortDirection) => {
-                if (!!direction) {
-                  setSorting({ ...sorting, [column]: direction });
-                } else {
-                  const data: any = { ...sorting };
-                  delete data[column];
-                  setSorting(data);
-                }
-              }}
-              canRowExpand={(val: any) => {
-                if (val.values.propertyTypeId === 0) {
-                  return true;
-                } else {
-                  return false;
-                }
-              }}
-              detailsPanel={{
-                render: val => {
-                  if (expandData[val.id]) {
-                    return <Buildings hideHeaders={true} data={expandData[val.id]} />;
+          canRowExpand={(val: any) => {
+            if (val.values.propertyTypeId === 0) {
+              return true;
+            } else {
+              return false;
+            }
+          }}
+          detailsPanel={{
+            render: val => {
+              if (expandData[val.id]) {
+                return <Buildings hideHeaders={true} data={expandData[val.id]} />;
+              }
+            },
+            icons: {
+              open: <FaFolderOpen color="black" size={20} />,
+              closed: <FaFolder color="black" size={20} />,
+            },
+            checkExpanded: (row, state) => !!state.find(x => checkExpanded(x, row)),
+            onExpand: loadBuildings,
+            getRowId: row => row.id,
+          }}
+          filter={appliedFilter}
+          onFilterChange={values => {
+            setFilter({ ...filter, ...values });
+          }}
+          renderBodyComponent={({ body }) => (
+            <Formik
+              innerRef={tableFormRef as any}
+              initialValues={{ properties: data || [] }}
+              enableReinitialize
+              onSubmit={async values => {
+                let nextProperties = [...values.properties];
+                const changedRows = dirtyRows.map(change => {
+                  const data = { ...values.properties![change.rowId] };
+
+                  return { rowId: change.rowId, data } as any;
+                });
+                if (changedRows.length > 0) {
+                  for (const change of changedRows) {
+                    const apiProperty = toApiProperty(change.data as any, true);
+                    const callApi = apiProperty.parcelId ? updateParcel : updateBuilding;
+                    const response: any = await callApi(apiProperty.id, apiProperty);
```

Comment:
does this perform ok if the user changes a relatively large number of rows at the same time? What happens if one of the requests fails?

Reply:
You need to handle if one or more of the requests fail

Reply:
Added a toast display and added row fields validation error display